contracts/src/token/erc721/mod.rs

```diff
@@ -1112,8 +1112,13 @@ impl Erc721 {
         let receiver = IERC721Receiver::new(to);
         let call = Call::new_in(self);
         let data = data.to_vec();
-        let result =
-            receiver.on_erc_721_received(call, operator, from, token_id, data);
+        let result = receiver.on_erc_721_received(
+            call,
+            operator,
+            from,
+            token_id,
+            data.to_vec().into(),
```

Comment:
I think we are doing an unnecessary conversion here. The function takes in `data` as Bytes, then converts it to a vec and now converts it back to Bytes. I think we can remove line 1114 and just pass in data here

Reply:
Alas, those are different `Bytes` types. One is from `alloy-primitives` and the other one from `stylus-sdk`

Reply:
Yeah, you correct @ggonzalez94. I've noticed the lane `let data = data.to_vec();`. `to_vec` method call is duplicated

Reply:
I see. Should we remove it an merge the PR? I think it is the only thing missing @qalisander @bidzyyys

Reply:
Adding this right now.